crates/forge-runner/src/running.rs

```diff
@@ -174,11 +174,18 @@ pub fn run_test_case(
 
     let initial_gas = usize::MAX;
     let runner_args: Vec<Arg> = args.into_iter().map(Arg::Value).collect();
+    let sierra_instruction_idx = case.test_details.sierra_entry_point_instruction_idx;
+    let (casm_entry_point_offset, actual_instruction_idx) =
+        casm_program.debug_info[sierra_instruction_idx];
+    assert_eq!(
+        actual_instruction_idx, sierra_instruction_idx,
+        "sierra_instruction_idx != actual_instruction_idx"
+    );
```

Comment:
Please don't rely on the vector being sorted, search for the relevant index instead.

Reply:
I wouldn't worry about the complexity, run instructions are always appended at the beginning of the instruction vec.